openedx/features/partners/g2a/views.py

```diff
@@ -1,9 +1,230 @@
+    EmailPreference,
+    Organization,
+    UserExtendedProfile
+)
+from lms.djangoapps.onboarding.models import RegistrationType
+from lms.djangoapps.philu_overrides.user_api.views import RegistrationViewCustom
+from nodebb.helpers import update_nodebb_for_user_status
+from notification_prefs.views import enable_notifications
+from openedx.core.djangoapps.lang_pref import LANGUAGE_KEY
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from openedx.core.djangoapps.user_api.preferences import api as preferences_api
+from openedx.core.djangoapps.user_authn.cookies import set_logged_in_cookies
+from openedx.core.djangoapps.user_authn.views.register import REGISTER_USER, \
+    record_registration_attributions
 from openedx.features.partners.helpers import get_partner_recommended_courses
+from pytz import UTC
+from student.models import Registration, UserProfile
+from util.json_request import JsonResponse
+
+from .forms import Give2AsiaAccountCreationForm
+
+log = logging.getLogger("edx.student")
+AUDIT_LOG = logging.getLogger("audit")
 
 
 def g2a_dashboard(request):
     # TODO: The argument must be dynamic after integration of LP-1632
     courses = get_partner_recommended_courses('give2asia')
     return render_to_response('partners/g2a/dashboard.html', {'recommended_courses': courses})
+
+
+class Give2AsiaRegistrationView(RegistrationViewCustom):
+    """
+    This class handles registration flow for give2asia users. It inherit some basic functionality from original (normal)
+    registration flow
+    """
+
+    def get(self, request, partner):
+        raise Http404
+
+    def post(self, request, partner):
+        registration_data = request.POST.copy()
+
+        # Adding basic data, required to bypass onboarding flow
+        registration_data['organization_name'] = partner.label
+        registration_data['year_of_birth'] = 2000
+        registration_data['level_of_education'] = 'IWRNS'
+
+        # validate data provided by end user
+        account_creation_form = Give2AsiaAccountCreationForm(data=registration_data, tos_required=True)
+        if not account_creation_form.is_valid():
+            return JsonResponse({"Error": dict(account_creation_form.errors.items())}, status=400)
+
+        account_creation_form.clean_registration_data(registration_data)
+        registration_data['first_name'], registration_data['last_name'] = registration_data['name'].split(" ", 1)
+
+        try:
+            # Create models for user, UserProfile and UserExtendedProfile
+            user = create_account_with_params_custom(request, registration_data)
+            self.save_user_utm_info(user)
+        except Exception:
+            error_message = {"Error": {"reason": "User registration failed"}}
+            log.exception(error_message)
+            return JsonResponse(error_message, status=400)
+
+        RegistrationType.objects.create(choice=1, user=request.user)
```

Comment:
Let's please not save Registration type because it was for split registration and we aren't using it. @emraan004 any thoughts on it?